### ISSUE TITLE
RUCSS - Resources aren't saved when WP_PLUGIN_DIR and WP_PLUGIN_URL don't match

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4437]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ `3.10.1`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

When the `WP_PLUGIN_DIR` and `WP_PLUGIN_URL` constants are altered and do not match:
```PHP
if ( ! defined( 'WP_PLUGIN_URL' ) )
       define( 'WP_PLUGIN_URL', 'http://www.example.com/ext' );
```
we are looking for plugin files using the wrong path, which results in not storing their content in the database:
https://github.com/wp-media/wp-rocket/blob/236b54e13b17adaa80f81f35803526608c8dcd03/inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcherProcess.php#L186

For example, the following resource:
https://www.example.com

isn't stored in:
`/ext/expand-divi/assets/styles/`

but in:
`/wp-content/plugins/expand-divi/assets/styles/`

In the end, the used CSS is missing used CSS selectors that exist in such files.

**To Reproduce**
1. Change the  `WP_PLUGIN_URL`.
2. Activate Remove Unused CSS.
3. Search the `wpr_rucss_resources` table for our `lazyload.min.js`. The file will be there, but the `content` will not contain the script.

**Expected behavior**

We should take into consideration `WP_PLUGIN_DIR` and `WP_PLUGIN_URL` constants when saving the resources in the database.

**Additional context**

**Ticket:** https://secure.helpscout.net/conversation/1661440621/301277/
**Slack convo:** https://wp-media.slack.com/archives/C029G1B5HD2/p1634288924026700

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by remyperona** [comment]
When using a different value for `WP_PLUGIN_URL` and the default one for `WP_PLUGIN_DIR`, I'm getting a 404 on all resources from the plugins.

Do you know if there is another customization done to make this work?

**Comment by engahmeds3ed** [comment]
Can't reproduce it till now.

I tried two cases:

1. Changing `WP_PLUGIN_URL` and keep `WP_PLUGIN_DIR` as is => it gives me 404 on all plugins' resources when viewing the page.
2. Changing both `WP_PLUGIN_URL` and `WP_PLUGIN_DIR` => this case is working properly and the resources' table has the content properly.

@wp-media/productrocket I think we need to exclude our lazyload script from the page, what do u think?

**Comment by piotrbak** [comment]
@vmanthos Were there any other steps that we'd need to follow? Per @Tabrisrp, if we have a different value for `WP_PLUGIN_URL` and the `WP_PLUGIN_DIR`, all the site is broken. Thanks!

**Comment by vmanthos** [comment]
@piotrbak I shared with @engahmeds3ed the configuration the customer had, but it didn't work for him.

I'll need to revisit this when I'm done with my backlog.

**Comment by piotrbak** [comment]
Fixed in 3.11 as we're not using the resources table anymore.
